modules/renter/repair.go

```diff
@@ -87,9 +86,19 @@ func (f *file) repair(chunkIndex uint64, missingPieces []uint64, r io.ReaderAt,
 			})
 			f.contracts[host.ContractID()] = contract
 			f.mu.Unlock()
+			errChan <- nil
 		}(missingPieces[i], hosts[i])
 	}
-	wg.Wait()
+	var errs []error
+	for i := 0; i < numPieces; i++ {
+		err := <-errChan
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	if len(errs) == numPieces {
+		return fmt.Errorf("could not upload any sectors:\n%v", build.JoinErrors(errs, "\n"))
```

Comment:
Even if not everyone had an error, it would be good to `log.Debugln` all of the other errors. For renters that are offline, this may create massive debug log files, but that's okay.